xarray/core/accessor_dt.py

```diff
@@ -304,6 +308,16 @@ def strftime(self, date_format):
             result, name="strftime", coords=self._obj.coords, dims=self._obj.dims
         )
 
+    def isocalendar(self):
+        "DataSet containing ISO year, week number, and weekday."
+
+        from .dataset import Dataset
+
+        if not is_np_datetime_like(self._obj.data.dtype):
+            raise AttributeError("'CFTimeIndex' object has no attribute 'isocalendar'")
+
+        return Dataset({"year": self.year, "week": self.week, "day": self.day})
```

Comment:
I think `isocalendar` returns the `weekday` instead of the day of the month (see [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Timestamp.isocalendar.html) in pandas, or [here](https://docs.python.org/3/library/datetime.html#datetime.date.isocalendar) in the standard library).
```suggestion
        return Dataset({"year": self.year, "week": self.week, "weekday": self.weekday})
```

Reply:
Thanks yes indeed. I find the pandas naming a bit unfortunate, then.